Movie Search/Models/DownloadOperation.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  DownloadOperation.swift
+//  Movie Search
+//
+//  Created by Alexey Grabik on 17/04/2020.
+//
+
+import UIKit
+
+class DownloadOperation: Operation {
+
+    private let movieModel: MovieModel
+    private let networkService: NetworkService
+    private let finishBlock: () -> Void
+    var downloadTaskDisposal: Disposable?
+    let indexPath: IndexPath
+    
+    private let semaphore = DispatchSemaphore(value: 0)
+    
+    init(indexPath: IndexPath,
+         movieModel: MovieModel,
+         networkService: NetworkService,
+         finishBlock: @escaping () -> Void) {
+        self.indexPath = indexPath
+        self.movieModel = movieModel
+        self.networkService = networkService
+        self.finishBlock = finishBlock
+        super.init()
+    }
+    
+    override func main() {
+        guard !isCancelled else {
+            return
+        }
+        
+        movieModel.photoRecord.state = .loading
+        
+        if let posterPath = movieModel.posterPath {
+            DispatchQueue.main.async {
```

Comment:
I'm not a fan of using a semaphore here. What's the purpose of blocking it until completion? It might indicate an operation is not the right type to use here. Also I'd opt to pass the result of your operation (the image) to the finish closure instead of updating your model directly from here.

Reply:
Thanks, but how can I tune simultaneous download tasks then?